lib/src/result/result.dart

```diff
@@ -36,6 +36,47 @@ sealed class Result<S, F extends Object> {
     }
   }
 
+  /// Similar to [Result()], but works in an asynchronous context.
+  ///
+  /// Creates a context for early return, similar to "Do notation". Works like the Rust "?" operator, which is a
+  /// "Early Return Operator". Here "$" is used as the "Early Return Key". when "$" is used on a type [Err],
+  /// immediately the context that "$" belongs to is returned with that [Err]. e.g.
+  ///
+  /// ```dart
+  ///     FutureResult<int,String> innerFn() async {
+  ///         return Err("message");
+  ///     }
+  ///
+  ///     FutureResult<int, String> innerFn2() async {
+  ///         return Ok(1);
+  ///     }
+  ///
+  ///     FutureResult<int, String> earlyReturn() => Result.earlyAsync(($) async {
+  ///         int y = 2;
+  ///         // the function will stop here since an Err was returned
+  ///         int x = await innerFn()[$];
+  ///         // innerFn2 will not be executed
+  ///         int z = await innerFn2()[$];
+  ///         return Ok(x + y + z);
+  ///       });
+  ///     }
+  ///     expect(await earlyReturn().unwrapErr(), "message");
+  ///```
+  ///
+  /// This should be used at the top level of a function as above. Passing "$" to any other functions, nesting, or
+  /// attempting to bring "$" out of the original scope should be avoided. Also, using the method [catchError] on the futures after
```

Comment:
Is this just  `_ResultEarlyReturnNotification<F>` will be caught inside the `Result` context, so a user should not expect `catchError` to catch that, or was there something else?

Reply:
Yes it's that. Since we use a try catch block to recieve the notification, the catchError function blocks this notification instead. You can test it by adding a catchError on a Err value and inspecting the error I believe.